FIAT/functional.py

```diff
@@ -406,6 +429,57 @@ def tostr(self):
         return "(u.t%d)(%s)" % (self.tno, ','.join(x),)
 
 
+class IntegralMomentOfFaceTangentEvaluation(Functional):
+    r"""
+    \int_F v \times n \cdot p ds
+
+    p \in Polynomials
+
+    :arg ref_el: reference element for which F is a codim-1 entity
+    :arg Q: quadrature rule on the face
+    :arg P_at_qpts: polynomials evaluated at quad points
+    :arg facet: which facet.
+    """
+    def __init__(self, ref_el, Q, P_at_qpts, facet):
+        P_at_qpts = [[P_at_qpts[0][i], P_at_qpts[1][i], P_at_qpts[2][i]]
+                     for i in range(P_at_qpts.shape[1])]
+        n = ref_el.compute_scaled_normal(facet)
+        sd = ref_el.get_spatial_dimension()
+        transform = ref_el.get_entity_transform(sd-1, facet)
+        pts = tuple(map(lambda p: tuple(transform(p)), Q.get_points()))
+        weights = Q.get_weights()
+        pt_dict = OrderedDict()
+        for pt, wgt, phi in zip(pts, weights, P_at_qpts):
+            phixn = [phi[1]*n[2] - phi[2]*n[1],
+                     phi[2]*n[0] - phi[0]*n[2],
+                     phi[0]*n[1] - phi[1]*n[0]]
+            pt_dict[pt] = [(wgt*(-n[2]*phixn[1]+n[1]*phixn[2]), (0, )),
+                           (wgt*(n[2]*phixn[0]-n[0]*phixn[2]), (1, )),
+                           (wgt*(-n[1]*phixn[0]+n[0]*phixn[1]), (2, ))]
+        super().__init__(ref_el, (sd, ), pt_dict, {}, "IntegralMomentOfFaceTangentEvaluation")
+
+
+class MonkIntegralMoment(Functional):
+    r"""
+    \int_F v\times n \cdot p ds where p \in P_{q-2}(f)^2
+
+    :arg ref_el: reference element for which F is a codim-1 entity
+    :arg Q: quadrature rule on the face
+    :arg P_at_qpts: polynomials evaluated at quad points
+    :arg facet: which facet.
+    """
```

Comment:
Maybe add ref to the book section (and note the lack of area scaling necessary here?)